_posts/en/newsletters/2020-10-28-newsletter.md

```diff
@@ -0,0 +1,234 @@
+
+        Even though Mallory created this second payment, she can't
+        complete it because she doesn't know the preimage for the
+        hashlock.  Instead, she closes her channel with Bob with the
+        second payment still pending.  Because of a bug in LND, Bob will
+        use the preimage meant for the payment from Alice to settle the
+        payment routed through his node by Mallory.  This earns Bob the
+        routing fees Mallory paid, but it also reveals the preimage to
+        the pending HTLC Alice tried routing through Mallory.  With the
+        preimage, Mallory is able to claim the full amount of that
+        payment without routing any part of it to Bob.  Additionally,
+        Alice receives cryptographic proof of payment even though
+        the payment was actually stolen.
+
+        LND's design tried to prevent this problem by storing payment
+        preimages separately from routing preimages, but some code was
+        added that queried both databases, creating the bug that Riard
+        discovered.
+
+        Both Fromknecht and Riard note that the attack could've been
+        prevented if the [payment secrets feature][] added to all LN
+        implementations almost a year ago had become mandatory
+        for all invoices.  This feature was designed to prevent
+        privacy-reducing probing, particularly of [multipath
+        payments][topic multipath payments],  by preventing a receiver
+        from disclosing a payment preimage unless a payment contained a
+        nonce included in the invoice.  If it had been enforced in this
+        case, Mallory wouldn't know the payment secret and so couldn't
+        have induced Bob to disclose it even with the buggy code.
+        Fromknecht notes that, "the upcoming v0.12.0-beta release of lnd
+        is likely to make payment secrets required by default.  We would
+        welcome other implementations to do the same."
+
+        A second set of emails from [Riard][riard6] and
+        [Fromknecht][fromknecht6] describe this issue in detail.
+
+## Selected Q&A from Bitcoin StackExchange
+
+*[Bitcoin StackExchange][bitcoin.se] is one of the first places Optech
+contributors look for answers to their questions---or when we have a
+few spare moments to help curious or confused users.  In
+this monthly feature, we highlight some of the top-voted questions and
+answers posted since our last update.*
+
+{% comment %}<!-- https://bitcoin.stackexchange.com/search?tab=votes&q=created%3a1m..%20is%3aanswer -->{% endcomment %}
+{% assign bse = "https://bitcoin.stackexchange.com/a/" %}
+
+- [Why is the Bitcoin Core wallet database moving from Berkeley DB to SQLite?]({{bse}}99620)
+  Michael Folkson references Andrew Chow's [blog post][achow wallet blog post]
+  including pros of SQLite, cons of Berkeley DB, and a bigger picture [proposed
+  timeline][wallet proposed timeline] to migrate from legacy to
+  [descriptor][topic descriptors] wallets.
```

Comment:
@harding do you think a topic reference here make sense / add value?

Reply:
I thought about it, but I think the mention of "descriptor wallets" in both Folkson's answer and your summary is really just the name of the wallet redesign project rather than a development related to descriptors.  Does that seem reasonable?

Reply:
Yep, reasonable. Resolving.